pysat/tests/test_utils_io.py

```diff
@@ -0,0 +1,381 @@
+
+        # Initalize the loaded data
+        self.loaded_inst = None
+        return
+
+    def teardown(self):
+        """Clean up the test environment."""
+
+        # Clear the attributes with data in them
+        del self.loaded_inst, self.testInst, self.stime
+
+        # Reset the pysat parameters
+        pysat.params['data_dirs'] = self.data_path
+
+        # Remove the temporary directory
+        self.tempdir.cleanup()
+
+        # Clear the directory attributes
+        del self.data_path, self.tempdir
+        return
+
+    def eval_loaded_data(self):
+        """Evaluate loaded test data."""
+        # Test that the written and loaded data matches the initial data
+        if self.testInst.pandas_format:
+            keys = self.testInst.data.columns
+            new_keys = self.loaded_inst.columns
+        else:
+            keys = [key for key in self.testInst.data.variables]
+            new_keys = [key for key in self.loaded_inst.variables]
+
+        # Test the data values for each variable
+        for dkey in keys:
+            lkey = dkey.lower()
+            if lkey in ['profiles', 'alt_profiles', 'series_profiles']:
+                # Test the loaded higher-dimension data
+                for tframe, lframe in zip(self.testInst.data[dkey],
+                                          self.loaded_inst[dkey]):
+                    assert np.all(tframe == lframe), "unequal {:s} data".format(
+                        dkey)
+            else:
+                # Test the standard data structures
+                assert np.all(self.testInst[dkey] == self.loaded_inst[dkey])
+        return keys, new_keys
+
+    def test_load_netcdf4_empty_filenames(self):
+        """Test raises ValueError without any filename input."""
+        with pytest.raises(ValueError) as verr:
+            pysat.utils.load_netcdf4(fnames=None,
+                                     pandas_format=self.testInst.pandas_format)
+
+        assert str(verr).find("Must supply a filename/list of filenames") >= 0
+        return
+
+    def test_basic_write_and_read_netcdf4_mixed_case_data_format(self):
+        """Test basic netCDF4 read/write with mixed case data variables."""
+        # Create a bunch of files by year and doy
+        outfile = os.path.join(self.testInst.files.data_path,
+                               'pysat_test_ncdf.nc')
+        self.testInst.load(date=self.stime)
+
+        # Modify data names in data
+        if self.testInst.pandas_format:
+            self.testInst.data = self.testInst.data.rename(str.upper,
+                                                           axis='columns')
+        else:
+            map_keys = {dkey: dkey.upper()
+                        for dkey in self.testInst.data.data_vars.keys()}
+            self.testInst.data = self.testInst.data.rename(map_keys)
+
+        # Meta case is preserved and has not been altered
+        self.testInst.to_netcdf4(outfile, preserve_meta_case=True)
```

Comment:
Should we switch to the new notation so we don't pile up deprecation warnings while running the tests?

Reply:
Yes, these are the tests of the future!  Let @JonathonMSmith handle the tests of the past 🦋